config/checkstyle_checks.xml

```diff
@@ -424,6 +424,14 @@
       <message key="matchxpath.match"
                value="Avoid using deprecated method 'DetailAst.branchContains()'."/>
     </module>
+    <module name="MatchXpath">
+      <property name="id" value="noUsageOfGetFileContents()Method"/>
+      <property name="query" value="//CLASS_DEF[
+              ./EXTENDS_CLAUSE[./IDENT[contains(@text,'Abstract') and contains(@text,'Check')]]
```

Comment:
>  in your case you need to make sure that classes are AbstractCheck to show violation

Why isn't this something like `@text = 'AbstractCheck'`?

Reply:
I explained in https://github.com/checkstyle/checkstyle/pull/9909#discussion_r622442239

For example, AbstractJavadocCheck extends AbstractCheck, and JavadocMetadataScraper extends AbstractJavadocCheck. Then JavadocMetadataScraper should not be allowed to use getFileContents()

Reply:
Then this is hiding things like:
````
$ ag "getFileContents"
src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheckTest.java
62:        assertSame(contents, check.getFileContents(), "expected file contents");

src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractCheckTest.java
223:        assertSame(fileContents, check.getFileContents(), "Invalid file contents");
````

Reply:
please use exact match by class name `/IDENT(equal(@text), 'AbstractCheck') or /IDENT(equal(@text), 'AbstractJavadocCheck') or `